specs/opcode/40BLOCKHASH.md

```diff
@@ -0,0 +1,29 @@
+# BLOCKHASH opcode
+
+## Procedure
+
+The `BLOCKHASH` opcode pops `block_number` off the stack and pushes the hash from chosen block. If the chosen block number is not in valid range (last 256 blocks, not including current block), it will push 0 onto the stack.
+
+## Constraints
+
+1. opId == 0x40
+2. State transition:
+    - gc + 2 (1 stack read + 1 stack write)
+    - stack_pointer + 0 (one pop and one push)
+    - pc + 1
+    - gas + 20
+3. Lookups: 3
+    - `block_number` is popped from the stack.
+    - `current_block_number` is in the block context table.
+    - `block_hash` is in the block context table.
+4. Additional Constraint
+    - `block_number` should be in valid range.
```

Comment:
This needs to be removed now

Reply:
Fixed. Thanks!

Reply:
Oops! yes of course it _should_ be included. When I saw the markdown file on a small screen, due to some indentation  issues I thought the `block_number` was included within `lookups` (hence I mentioned that this is not a lookup).

@jeong0982 since this is a constraint, yes it should be present in the spec doc. Thanks for noticing @CPerezz :)

Reply:
I was also confused. I will put it back. Thank you👍🏼